### PR TITLE
:sparkles: Board as ruler origin

### PR DESCRIPTION
**Pull request by superalex**
Related to https://tree.taiga.io/project/penpot/task/5142

### CODE REVIEW

File: frontend/src/app/main/ui/workspace/viewport.cljs [hunk 1]
```diff
@@ -230,9 +231,23 @@
...
-                          (= (:layout selected-frame) :flex)
-                          (zero? (:rotation (first selected-shapes))))]
+                       (= (count selected-shapes) 1)
+                       (= (:layout selected-frame) :flex)
+                       (zero? (:rotation (first selected-shapes))))
+
+
+        selecting-first-level-frame? (and (= (count selected-shapes) 1)
+                                       (= (:type (first selected-shapes)) :frame)
+                                       (= uuid/zero (:frame-id (first selected-shapes))))
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/viewport.cljs]: better to use `cph/root-frame?`

File: frontend/src/app/main/ui/workspace/viewport.cljs [hunk 2]
```diff
@@ -230,9 +231,23 @@
...
+
+        offset-x (if selecting-first-level-frame?
+                   (:x (first selected-shapes))
```
Comment (Alotor) [inline frontend/src/app/main/ui/workspace/viewport.cljs]: you're using all the time `(first selected-shapes)` maybe it's better to put it in a variable

File: frontend/src/app/main/ui/workspace/viewport.cljs (hunk 1 above)
Comment (superalex) [reply frontend/src/app/main/ui/workspace/viewport.cljs]: Thanks!

File: frontend/src/app/main/ui/workspace/viewport.cljs (hunk 2 above)
Comment (superalex) [reply frontend/src/app/main/ui/workspace/viewport.cljs]: Thanks!

**Pull request merged**